cli/cmd/upgrade_legacy.go

```diff
@@ -75,6 +75,13 @@ func loadStoredValuesLegacy(ctx context.Context, k *k8s.KubernetesAPI) (*charts.
 				return nil, fmt.Errorf("values subpath not found in %s configmap", k8s.AddOnsConfigMapName)
 			}
 
+			// repair Add-On configs
+			repairedCm, err := repairAddOnConfig([]byte(cmData))
+			if err == nil {
```

Comment:
I think this only returns an error if there is an error during marshalling or unmarshalling.  In that case, should we log the error and/or return it?

Reply:
Logging seems right, as returning fails the installation flow, which seems not necessary for a Add-On fix.